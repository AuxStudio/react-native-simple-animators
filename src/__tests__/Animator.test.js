import React from 'react';
import renderer from 'react-test-renderer';
import { Animated } from 'react-native';

import Animator from '../';

describe('EventAnimator', () => {
  it('renders an EventAnimator when animatedValue was passed in', () => {
    expect(
      renderer.create(
        <Animator
          type="height"
          animatedValue={new Animated.Value(0)}
          interpolation={{ inputRange: [0, 1], outputRange: [0, 100], extrapolate: 'clamp' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders a StaticAnimator when animatedValue was not passed in', () => {
    expect(
      renderer.create(
        <Animator
          type="height"
          initialValue={0}
          finalValue={1}
          shouldAnimateIn
          animateInCallback={jest.fn()}
          easing={jest.fn()}
        />,
      ),
    ).toMatchSnapshot();
  });
});
