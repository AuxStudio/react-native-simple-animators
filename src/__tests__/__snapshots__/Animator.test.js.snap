// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a EventAnimator from Animator when animatedValue was passed in 1`] = `
<View
  style={
    Object {
      "height": 0,
    }
  }
/>
`;

exports[`renders a StaticAnimator from Animator when animatedValue was not passed in 1`] = `
<View
  style={
    Object {
      "height": 0,
    }
  }
/>
`;
